### ISSUE TITLE
Multiple distinct default values error incorrectly reported

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#69078]
**Version Used**: Default on SharpLab

**Steps to Reproduce**:

```cs
using System.Runtime.InteropServices;
class C
{
    static void M([DefaultParameterValue(1)] int i = 1) { }
}
```

[SharpLab](https://sharplab.io/#v2:D4AQDABCCMB0BKBXAdgFwJYFsCmsCSa2ATgPYAOAysQG7oDG2AzgNwCwAUCAEwQDCHAbw4QRUaADYoAFggBZABQBtACLYAZgENEAG1QAFDUQ05UxAGobtibPOgBKALoR0aZxAC8EexAEQAvhx+QA)

**Expected Behavior**:
- error CS1745: Cannot specify default parameter value in conjunction with DefaultParameterAttribute or OptionalAttribute

**Actual Behavior**:
- error CS1745: Cannot specify default parameter value in conjunction with DefaultParameterAttribute or OptionalAttribute
- error CS8017: The parameter has multiple distinct default values.

See https://github.com/dotnet/roslyn/pull/69032#discussion_r1265334199 for details.
